src/system/SystemSockets.h

```diff
@@ -26,18 +26,50 @@
 // Include configuration headers
 #include <system/SystemConfig.h>
 
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS
+
+#include <support/BitFlags.h>
 #include <system/SystemError.h>
 
 namespace chip {
 namespace System {
 
 using ::chip::System::Error;
 
+enum class SocketEventFlag : uint8_t
+{
+    kRead  = 0x01, /**< Bit flag indicating if there is a read event on a socket. */
+    kWrite = 0x02, /**< Bit flag indicating if there is a write event on a socket. */
+    kError = 0x04, /**< Bit flag indicating if there is an error event on a socket. */
+};
+
+using SocketEvents = BitFlags<SocketEventFlag>;
+
+class SystemSocketWatcherBasis
+{
+public:
+    typedef void (*Callback)(int socket, SocketEvents events, void * data);
```

Comment:
We could replace this class with

`using SystemSocketWatcherCallback = void(int socket, SocketEvents events, void * data);` 

?

Reply:
Actually, after further investigation, I do want to have a base class containing the file descriptor and active-event flags — essentially, changing the consumers' members from `int mSocket; SocketEvents mPendingIO; SystemSocketWatcher mWatcher;` as here, to `WatchableSocket mSocket`.